File: 242-valid-anagram/valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length){
            return false;
        }
        Dictionary<char,int> frequ = new Dictionary<char,int>();
        foreach(char c in s){
            if(frequ.ContainsKey(c)) {
                frequ[c]++;
            }else{
                frequ[c] = 1;
            }
        }

        foreach(char c in t){
            if(!frequ.ContainsKey(c) || frequ[c] == 0) { return false;}
            frequ[c]--;
        }

        return true;
    }
}

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        unordered_map<char,int> freq;

        if(s.length()!=t.length()) return false;
        for(int i = 0 ;i<s.length(); i ++){
            if(freq.find(s[i]) != freq.end() ){
                freq[s[i]]++;
            }else{
                freq[s[i]] = 1;
            }
        }


        for(int i = 0; i<t.length(); i++){
            if(freq.find(t[i]) == freq.end() || freq[t[i]] == 0 ){
                return false;
            }else{
                freq[t[i]] --;
            }

        }


        return true;
    }
};